src/app/common/services/confirm-save.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+
+import { Observable, of } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+import { IConfirmSave } from '../models/Confirm-save';
+
+@Injectable()
+export class ConfirmSaveService<T> {
+  public confirmNavigation(component: T, config: IConfirmSave<T>): Observable<boolean> {
+    if (config.checkEmpty(component)) {
+      return of(true);
+    }
+
+    const { message, disable } = config;
```

Comment:
better then to destructure right in the function 
`public confirmNavigation(component: T, { message, disable }: IConfirmSave<T>): Observable<boolean> {`